scheduler/select.go

```diff
@@ -3,18 +3,24 @@ package scheduler
 // LimitIterator is a RankIterator used to limit the number of options
 // that are returned before we artificially end the stream.
 type LimitIterator struct {
-	ctx    Context
-	source RankIterator
-	limit  int
-	seen   int
+	ctx              Context
+	source           RankIterator
+	limit            int
+	maxSkip          int
+	scoreThreshold   float64
+	seen             int
+	skippedNodes     []*RankedNode
+	skippedNodeIndex int
 }
 
 // NewLimitIterator is returns a LimitIterator with a fixed limit of returned options
-func NewLimitIterator(ctx Context, source RankIterator, limit int) *LimitIterator {
+func NewLimitIterator(ctx Context, source RankIterator, limit int, scoreThreshold float64, maxSkip int) *LimitIterator {
 	iter := &LimitIterator{
-		ctx:    ctx,
-		source: source,
-		limit:  limit,
+		ctx:            ctx,
+		source:         source,
+		limit:          limit,
+		maxSkip:        maxSkip,
+		scoreThreshold: scoreThreshold,
```

Comment:
Is skipping going to be common? If so we should probably preallocate the slice to its max size. It won't use much memory but will save a lot of allocations since every append that grows a slice does a new allocation.

Reply:
Will make this change because it is the right way to do it, but was curious - does it matter if the slice will only grow to a max of 5 items?

Reply:
> does it matter if the slice will only grow to a max of 5 items?

Seems unlikely.